lms/static/js/groups/views/cohort_editor.js

```diff
@@ -206,35 +213,81 @@
-                    }
-                    else if (this.confirmationNotifications) {
+                    } else if (this.confirmationNotifications) {
                         this.confirmationNotifications.$el.html('');
                         this.confirmationNotifications = null;
                     }
 
+                    // Show preassigned email addresses.
+                    this.undelegateViewEvents(this.preassignedNotifications);
+                    if (numPreassigned > 0) {
+                        details = [];
+                        for (i = 0; i < modifiedUsers.preassigned.length; i++) {
+                            details.push(interpolate_text(gettext('{email}'),
+                                {email: modifiedUsers.preassigned[i]}));
+                        }
+
+                        title = (
+                            interpolate_text(
+                                ngettext('{numPreassigned} learner was pre-assigned for this cohort. ' +
+                                    'This learner will automatically be added to the cohort when ' +
+                                    'they enroll in the course.',
+                                    '{numPreassigned} learners were pre-assigned for this cohort. ' +
+                                    'These learners will automatically be added to the cohort when ' +
+                                    'they enroll in the course.',
+                                    numPreassigned),
+                                {numPreassigned: numPreassigned}
+                            )
+                        );
+
+                        this.preassignedNotifications = new NotificationView({
+                            el: this.$('.cohort-preassigned'),
+                            model: new NotificationModel({
+                                type: 'warning',
+                                title: title,
+                                details: details
+                            })
+                        });
+                        this.preassignedNotifications.render();
+                    } else if (this.preassignedNotifications) {
+                        this.preassignedNotifications.$el.html('');
+                        this.preassignedNotifications = null;
+                    }
+
                     // Show error messages.
                     this.undelegateViewEvents(this.errorNotifications);
-                    numErrors = modifiedUsers.unknown.length;
+                    numErrors = modifiedUsers.unknown.length + modifiedUsers.invalid.length;
                     if (numErrors > 0) {
-                        createErrorDetails = function(unknownUsers, showAllErrors) {
-                            var numErrors = unknownUsers.length, details = [];
+                        createErrorDetails = function(unknownUsers, invalidEmails, showAllErrors) {
+                            var unknownErrorsShown = showAllErrors ? unknownUsers.length :
```

Comment:
Oh interesting-- I forgot that we added this logic so that by default we don't show all learners that produced an error (we show the total number, the first 5, and a button to show all of them).

![image](https://user-images.githubusercontent.com/484484/27260776-7d99393e-5402-11e7-91bd-29a2168fe42d.png)

@sstack22 if you think that people will preassign a lot of learners at once through the UI (not the CSV upload), we could add this same functionality for preassigned learners. What do you think?